backend/src/test/java/com/google/minicrm/data/LeadTest.java

```diff
@@ -0,0 +1,156 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.minicrm.data;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.User;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.io.File;
+import java.io.FileReader;
+import java.io.Reader;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Provides Unit tests for the Lead data object.
+ */
+@RunWith(JUnit4.class)
+public final class LeadTest {
+
+  private static final File leadFile = new File("src/test/resources/lead1.txt");
+  private static final String TEST_USER_ID = "testUserId";
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void leadFromReader_fromValidLeadJson_producesEquivalentValidLead() throws Exception {
+    User testUser = new User("email", "authDomain", TEST_USER_ID);
+    Key advertiserKey = Advertiser.generateKey(testUser);
+    Reader reader = new FileReader(leadFile);
```

Comment:
maybe just add comment here to clarify that this is where the test property values are coming from.

Reply:
added comment here and to the leadFile instance variable at the top of the class